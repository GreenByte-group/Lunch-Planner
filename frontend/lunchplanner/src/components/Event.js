import React from "react";
//import PropTypes from "prop-types";

function Event(props) {
    return (
        <div className="event">
            <span>{props.name}</span>
        </div>
    );
}

// Typüberprüfung
//Event.propTypes = {
//    name: PropTypes.string.isRequired
//};

export default Event;