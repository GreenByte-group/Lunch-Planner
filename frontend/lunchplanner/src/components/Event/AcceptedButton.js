import React from "react"
import {Done} from "@material-ui/icons";
import {withStyles} from "material-ui";

const styles = theme => ({
    root: {
        float: 'right',
    },
    button: {
        border: '2px solid',
        borderRadius: '50%',
        height: '32px',
        width: '32px',
        padding: '3px',
        marginLeft: 'auto',
        marginRight: 'auto',
        display: 'block',
        float: 'right',
    },
    text: {
        fontFamily: "Work Sans",
        fontSize: '11px',
        fontWeight: '500',
        lineHeight: '16px',
        color: theme.palette.primary.main,
        marginLeft: 'auto',
        marginRight: '11px',
        float: 'left',
    },
});

class AcceptedButton extends React.Component {
    render() {
        const {classes} = this.props;
        const {text} = this.props;

        let styleButton = {};
        if(text)
            styleButton = {height: '12px', width:'12px', padding: '1px', borderWidth: '1px', marginRight: '16px', marginTop: '3px', marginBottom: '11px'};

        return (
            <div className={classes.root}>
                {(text
                        ? <span className={classes.text} >{text}</span>
                        : ""
                )}
                <Done style={styleButton} color="primary" className={classes.button} />
            </div>
        )
    }
}

export default withStyles(styles)(AcceptedButton);