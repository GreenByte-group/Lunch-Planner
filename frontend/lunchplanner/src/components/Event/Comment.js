import React from "react"
import moment from "moment"
import ListItem from "material-ui/List/ListItem";
import {withStyles} from "material-ui";

const styles = {
    root: {
      width: '100%',
    },
    text: {
        width: '100%',
        float: 'none',
    },
    dateText: {
        width: '100%',
        float: 'none',
    },
};

class Comment extends React.Component {

    constructor(props) {
        super();

        //http://momentjs.com/docs/
        let date = moment(props.date);

        this.state = {
            rightAlign: props.rightAlign,

            text: props.text,
            dateText: date.format("HH:mm"),
            username: props.creater,
        }
    }

    render() {
        const {classes} = this.props;
        let textAlign;
        let dateAlign;

        if(this.state.rightAlign) {
            textAlign = 'right';
            dateAlign = 'left';
        } else {
            textAlign = 'left';
            dateAlign = 'right';
        }


        return (
            <ListItem button className={classes.listItem}>
                <div className={classes.root}>
                    <p className={classes.text} style={{textAlign: textAlign}}>
                        {this.state.text}
                    </p>
                    <p className={classes.dateText} style={{textAlign: dateAlign}}>
                        {this.state.username}, {this.state.dateText}
                    </p>
                </div>
            </ListItem>
        );
    }
}

export default withStyles(styles)(Comment);