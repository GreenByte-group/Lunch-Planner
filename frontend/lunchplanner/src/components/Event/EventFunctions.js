import axios from "axios";
import {HOST} from "../../Config";
import moment from "moment/moment";

export function getEvents(search, responseFunc) {
    let url = HOST + "/event";
    if(search)
        url = HOST + "/event/search/" + search;

    axios.get(url)
        .then(responseFunc);
}

export function getEvent(eventId, responseFunc) {
    let url = HOST + "/event/" + eventId;

    axios.get(url)
        .then(responseFunc);
}

export function replyToEvent(eventId, answer, responseFunc) {
    let config = {
        headers: {
            'Content-Type': 'text/plain',
        }
    };

    let url = HOST + '/event/' + eventId + '/reply';
    axios.put(url, answer, config)
        .then(responseFunc)
}

export function changeEventTitle(eventId, title, responseFunc, errorFunc) {
    let config = {
        headers: {
            'Content-Type': 'text/plain',
        }
    };

    let url = HOST + '/event/' + eventId + '/name';
    axios.put(url, title, config)
        .then(responseFunc)
        .catch(errorFunc)
}

export function changeEventLocation(eventId, location, responseFunc, errorFunc) {
    let config = {
        headers: {
            'Content-Type': 'text/plain',
        }
    };

    let url = HOST + '/event/' + eventId + '/location';
    axios.put(url, location, config)
        .then(responseFunc)
        .catch(errorFunc)
}

export function changeEventTime(eventId, time, responseFunc, errorFunc) {
    let config = {
        headers: {
            'Content-Type': 'text/plain',
        }
    };

    let url = HOST + '/event/' + eventId + '/timestart';
    axios.put(url, time.valueOf(), config)
        .then(responseFunc)
        .catch(errorFunc)
}

export function createEvent(location, date, member, visible, locationId, responseFunc, errorFunc) {
    let momentDate = moment(date);

    let timeEnd = date.getTime();
    let url =  HOST + '/event';


    let data = {name: location, description: "", location : location, timeStart: date, visible: visible};
    if(locationId)
        data = {name: location, description: "", location : location, timeStart: date, visible: visible, locationId: locationId};

    console.log('function', data);
    axios.post(url, data)
        .then((response) => {
            if(response.status === 201) {
                inviteMemberToEvent(response.data, member);
            }
            responseFunc(response);
        })
        .catch(errorFunc);
}

export function inviteMemberToEvent(eventId, member, responseFunc, errorFunc) {
    if(member && member instanceof Array)

    member.forEach((oneMember) => {
        let url = HOST + "/event/" + oneMember + "/invite/event/" + eventId;
        axios.post(url)
            .then((response) => {
                if(response.status === 201)
                    if(responseFunc)
                        responseFunc(oneMember)
            })
    });
}

export function getEventExtern(token,responseFunc) {
    let url = HOST + "/event/token/" + token;
    axios.get(url)
        .then(responseFunc)
}