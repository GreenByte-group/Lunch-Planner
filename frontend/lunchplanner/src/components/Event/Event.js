import React from "react"
import moment from "moment"
import ListItem from "material-ui/List/ListItem";
import {withStyles} from "material-ui";
import {Schedule, Today} from "@material-ui/icons";
import AcceptedButton from "./AcceptedButton";
import {Link} from "react-router-dom";
import EventScreen from "./EventScreen";

const styles = {
    listItem: {
        padding: '7px 16px',
        '&:hover': {
            backgroundColor: '#0303031a !important',
        }
    },
    title: {
        fontFamily: "Work Sans",
        fontSize: '16px',
        lineHeight: '24px',
        marginBottom: '0px',
    },
    date: {
        fontFamily: "Work Sans",
        fontSize: '14px',
        lineHeight: '20px',
        marginBottom: '0px',
    },
    users: {
        fontFamily: "Work Sans",
        fontSize: '13px',
        lineHeight: '20px',
        marginBottom: '0px',
        color: '#A4A4A4',
    },
    icons: {
        width: '13px',
        height: 'auto',
    },
    text: {
        float: 'left',
        width: '100%',
    }
};

class Event extends React.Component {

    constructor(props) {
        super();

        //http://momentjs.com/docs/
        let date = moment(props.date);

        this.state = {
            background: props.background,
            accepted: props.accepted | false,
            id: props.id,
            name: props.name,
            description: props.description,
            monthDay: date.format('DD MMM'),
            time: date.format('HH:mm'),
            location: props.location,

            //TODO invited people
            people:
                [{userName: "Can", short: "C"},
                  {userName: "Felix", short: "F"},
                  {userName: "Martin", short: "M"},
                ],
        }
    }


    render() {
        const {classes} = this.props;

        const background = this.state.background;
        let accepted= this.state.accepted;

        let name = this.state.name;
        let description = this.state.description;
        let monthDay = this.state.monthDay;
        let time = this.state.time;
        let people = this.state.people;
        let location = this.state.location;

        return (
            <Link to={{pathname:`/event/${this.state.id}`, query:{
                    source: "/event",
                    eventName: name,
                    description: description,
                    monthDay: monthDay,
                    time: time,
                    people: people,
                    accepted: accepted,
                    location:location
                }}}>
                <ListItem style={{backgroundColor: background}} button className={classes.listItem}>
                    <div className={classes.text}>
                        <p className={classes.title}>{name}</p>
                        <p className={classes.date}><Today viewBox="-5 -5 27 27" className={classes.icons} /> {monthDay} <Schedule viewBox="-5 -5 27 27" className={classes.icons}/> {time}</p>
                        <p className={classes.users}>People</p>
                    </div>
                    {(accepted
                            ? <AcceptedButton text="Accepted" />
                            : ""
                    )}
                </ListItem>
            </Link>

        );
    }
}

export default withStyles(styles)(Event);