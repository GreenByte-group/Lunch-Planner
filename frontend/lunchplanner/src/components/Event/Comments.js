import React from "react";
import {List, ListItem, TextField, withStyles} from "material-ui";
import {HOST} from "../../Config";
import axios from "axios/index";
import {getUsername, setAuthenticationHeader} from "../authentication/Authentication";
import Comment from "./Comment";

const styles = {
    list: {
        padding: '0px',
    },
    textField: {
        width: '100%',
        padding: '16px',
    }
};

class Comments extends React.Component {

    constructor(props) {
        super();

        setAuthenticationHeader();

        this.state = {
            eventId: props.eventId,
            comments: [],
            newComment: "",
        }
    }

    componentDidMount() {
        let url = HOST + "/event/" + this.state.eventId + "/getComments";

        axios.get(url)
            .then((response) => {
                this.setState({
                    comments: response.data,
                })
            })
      
        // this.setState({
        //     comments: [{userName: "Martin", date: new Date(), commentText: "Text 1 a bilt longer than normal"},
        //         {userName: "Felix", date: new Date(), commentText: "Text 1 a bilt longer than normal"},
        //         {userName: "Can", date: new Date(), commentText: "Text 1 a bilt longer than normal"},
        //         {userName: "Martin", date: new Date(), commentText: "Text 1 a bilt longer than normal"}]
        // })
    }

    textFieldChanged = (event) => {
        this.setState({
            newComment: event.target.value,
        })
    };

    onSubmit = (event) => {
      console.log(this.state.newComment);

      let url = HOST + "/event/" + this.state.eventId + "/comment";

      let config = {
          headers: {
              'Content-Type': 'text/plain',
          }
      };

      axios.put(url, this.state.newComment, config)
          .then((response) => {
              console.log(response)
          });

      event.preventDefault();
    };

    render() {
        const {classes} = this.props;
        const comments = this.state.comments;

        return (
            <div>
                <form
                    className={classes.textFieldComment}
                    noValidate autoComplete="on"
                    onSubmit={this.onSubmit}
                >
                    <TextField
                        className={classes.textField}
                        id="textFieldComment"
                        value={this.state.newComment}
                        onChange={this.textFieldChanged}
                        placeholder ="Write a comment ..."

                    />
                </form>
                <List className={classes.list}>
                    {comments.map((listValue) => {
                        let rightAlign = false;
                        if(listValue.userName === getUsername())
                            rightAlign = true;

                        return (
                            <Comment
                                text={listValue.commentText}
                                date={listValue.date}
                                creater={listValue.userName}
                                rightAlign={rightAlign}
                            />
                        )
                    })}
                </List>
            </div>
        )
    }

}

export default withStyles(styles)(Comments);