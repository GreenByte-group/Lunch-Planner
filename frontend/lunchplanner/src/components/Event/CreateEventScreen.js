import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Dialog from '../Dialog';
import {Switch, Typography, TextField, ExpansionPanel, ExpansionPanelSummary, ExpansionPanelDetails} from '@material-ui/core';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import AddIcon from '@material-ui/icons/Add';
import MapIcon from '@material-ui/icons/Map'
import VisibilityOffIcon from '@material-ui/icons/VisibilityOff';
import VisibilityIcon from '@material-ui/icons/Visibility';
import {FormGroup, FormControlLabel, Slide} from '@material-ui/core';
import {DatePicker, TimePicker} from 'material-ui-old';
import PeopleIcon from '@material-ui/icons/People'

import 'react-datepicker/dist/react-datepicker-cssmodules.css';
import "../../assets/CreateEventScreen.css"
import {Link} from "react-router-dom";
import {Today, Schedule} from "@material-ui/icons";
import {eventListNeedReload} from "./EventList";
import moment from "moment";

import {getHistory} from "../../utils/HistoryUtils";
import {InputAdornment} from "@material-ui/core";
import {createEvent} from "./EventFunctions";
import GoogleSuggest from "../Map/GoogleSuggest";
import Icon from "@material-ui/core/es/Icon/Icon";

const styles = {
    appBar: {
        position: 'relative',
    },
    flex: {
        flex: 1,
    },
    textField: {
        marginBottom:30,
        marginLeft: 20,
        width: "60%",
    },
    searchfield: {
        marginBottom:30,
        marginLeft:20,
        width:"100%",
        float: "left"
    },
    searchboxField: {
      marginTop:30,
        width:"55%",
        float: "left"

    },
    mapIcon:{
        marginLeft:'15px',
        height:'10px',
        width:'50%'
    },
    button:{
        fontSize: '16px',
        fontFamily: 'Work Sans',
        color: "white",
        position: "fixed",
        bottom: 0,
        width: "100%",
        height: '56px',
        zIndex: '10000',
    },
    error: {
        textAlign: 'center',
        color: '#ff7700',
        marginTop: '10px',
        marginBottom: '0px',
    },
    pickerWithIcon: {
        width: '50%',
        float: 'left',
    },
    mapIcon: {
        width: '60% !important',
        overflow: 'hidden',
        float: 'right'
    },
    visibilityIcon:{
        color:'#D3D3D3',
    },
    datePicker: {
        width: '60% !important',
        overflow: 'hidden',
        float: 'left',
    },
    timePicker: {
        width: '60% !important',
        overflow: 'hidden',
        float: 'left',
    },
    pickerTextField: {
        fontSize: '14px !important',
        height: '35px !important',
        widht: 'auto',
        lineHeight: '34px',
    },
    dateHeader: {
        paddingLeft: '20px',
        marginBottom: '0px',
        float: 'left',
        width: '50%',
        fontSize: '11px',
        color: '#A4A4A4',
    },
    timeHeader: {
        float: 'left',
        width: '50%',
        marginBottom: '0px',
        fontSize: '11px',
        color: '#A4A4A4',
    },
    icons: {
        marginTop: '4px',
        marginRight: '5px',
        width: '20px',
        height: 'auto',
        float: 'left',
        color: '#A4A4A4',
    },

    inviteTextField: {
        width: '100%',
    },
    overButton: {
        height: '100%',
        overflowY: 'auto',
        display: 'flex',
        flexDirection: 'column',
    },
};
const buttonStyle = {
    float: 'right',
    marginBottom: '15px',
};

function Transition(props) {
    return <Slide direction="up" {...props} />;
}


class CreateEventScreen extends React.Component {

    constructor(props) {
        super();
        const params = new URLSearchParams(props.location.search);

        let defaultDate = moment().add(30, 'm').toDate();

        this.state = {
            open: true,
            name: params.get('name') || "",
            visible: params.get('visible') || false,
            date: params.get('date') || defaultDate,
            invitedUsers: params.get('invitedUsers') || [],
            invitedTeams: params.get('invitedTeams') || [],
            invitedTeamMember: params.get('teamMember') || [],
            location: params.get('location') || "",
            error: "",
        };
    }

    parseUrl = () => {
        const params = new URLSearchParams(this.props.location.search);
        let invitedUsers = params.get('invitedUsers');
        let invitedTeams = params.get('invitedTeams');
        let teamMember = params.get('teamMember');
        if(invitedUsers != null && invitedUsers !== undefined && invitedUsers !== this.state.invitedUsers) {
            this.setState({
                invitedUsers: params.get('invitedUsers'),
            });
        }
        if(invitedTeams != null && invitedTeams !== undefined && invitedTeams !== this.state.invitedTeams) {
            this.setState({
                invitedTeams: params.get('invitedTeams'),
            });
        }
        if(teamMember != null && teamMember !== undefined && teamMember !== this.state.invitedTeamMember) {
            this.setState({
                invitedTeamMember: params.get('teamMember'),
            });
        }
    };

    handleAccept = () => {
        let created = this.state.created;

        let invitedUsers = "";
        if(this.state.invitedUsers)
            invitedUsers += this.state.invitedUsers;

        if(this.state.invitedTeamMember) {
            if(invitedUsers)
                invitedUsers += ",";
            invitedUsers += this.state.invitedTeamMember;
        }

        let invitedUsersArray = [];
        if(invitedUsers)
            invitedUsersArray = invitedUsers.split(",")

        createEvent(this.state.location, this.state.date, invitedUsersArray, this.state.visible,
            (response) => {
                if(response.status === 201) {
                    eventListNeedReload();
                    getHistory().push('/event');
                } else {
                    this.setState({error: response.response.data});
                }
            },
            (error) => {
                if(error && error.response)
                    this.setState({error: error.response.data});
                else
                    console.log(error);
            });
    };

    handleClickOpen = () => {
        this.setState({ open: true });
    };

    handleChange = (event) => {
        let target = event.target;
        this.setState({
            [target.id]: target.value,
        });
    }

    handleDate = (event, date) => {
        let newDate = moment(date);
        let dateBefore = moment(this.state.date);
        newDate.hour(dateBefore.hour());
        newDate.minute(dateBefore.minute());

        this.setState({ date: newDate.toDate() });
    }

    handleTime = (event, date) => {
        this.setState({ date: date });
    }

    handleVisibility = name => event =>{
        this.setState({ [name]: event.target.checked });
    }
    checkVisibility = () => {
        if(this.state.visible){
            return true;
        }
    }
    handleLocationChange= (location) => {
        this.setState({
            location: location,
        });
        console.log("Location: ", location);
        console.info("l√§ngengrad: "+location.l)
    }

    render() {
        this.parseUrl();
        const { classes } = this.props;
        const error = this.state.error;
        let invited = this.state.invitedUsers + "," + this.state.invitedTeams;
        let buttonEnabled = false;
        if(this.state.location)
            buttonEnabled = true;

        return (
            <Dialog
                title="Create Event"closeIconAbsolute
                closeUrl="/event"
                paddingBottom={'48px'}
            >
                <div className={classes.overButton}>
                    {(error
                            ? <p className={classes.error}>{error}</p>
                            : ""
                    )}
                    {/*<form noValidate autoComplete="on" >*/}
                       <FormGroup row>

                           <GoogleSuggest
                               className={classes.searchboxField}
                               id="location"
                               label="Location"
                               value={this.state.location}
                               placeholder="Add an Location ..."
                               float="left"
                               onChange={this.handleLocationChange}
                           />

                        <Link className={classes.mapIcon}
                              float="right"
                              to={{pathname: "/event/create/map", query: {
                                  location: this.state.location,}}}
                              location={ this.state.location}
                        >   <MapIcon disabled={false} className={classes.mapIcon} />
                        </Link>
                       </FormGroup>
                    {/*</form>*/}
                    <div>
                        <p className={classes.dateHeader}>Date</p><p className={classes.timeHeader}>Time</p>
                        <div className={classes.pickerWithIcon}>
                            <Today viewBox="-2 -4 26 26" className={classes.icons} style={{marginLeft: '18px'}} />
                            <DatePicker
                                className={classes.datePicker}
                                onChange={this.handleDate}
                                value={this.state.date}
                                textFieldStyle={styles.pickerTextField}
                            />
                        </div>
                        <div className={classes.pickerWithIcon}>
                            <Schedule viewBox="-2 -4 26 26" className={classes.icons}/>
                            <TimePicker
                                className={classes.timePicker}
                                onChange={this.handleTime}
                                value={this.state.date}
                                format="24hr"
                                textFieldStyle={styles.pickerTextField}
                            />
                        </div>
                    </div>
                    <ExpansionPanel>
                        <ExpansionPanelSummary expandIcon={<ExpandMoreIcon color='primary' />}>
                            <Typography className={classes.heading}>Invite Somebody & Incognito Mode </Typography>
                        </ExpansionPanelSummary>
                        <ExpansionPanelDetails>
                            <FormGroup row>
                                <Link className={classes.inviteTextField}
                                      to={{pathname: "/event/create/invite",  query: {
                                        source: "/event/create",
                                        invitedUsers: this.state.invitedUsers,
                                    }}}>
                                    <TextField
                                        InputLabelProps={{
                                            shrink: true,
                                        }}
                                        id="invitation"
                                        label="Participants"
                                        placeholder ="Invite People"
                                        value={invited}
                                        fullWidth
                                        InputProps={{
                                            startAdornment: <InputAdornment position="start"><PeopleIcon/></InputAdornment>,
                                            endAdornment: <InputAdornment position="end">
                                                <AddIcon />
                                            </InputAdornment>
                                        }}
                                    />
                                </Link>

                                <FormControlLabel
                                    control={
                                        <Switch
                                            float ="left"
                                            color = "primary"
                                            checked={this.state.visible}
                                            onChange={this.handleVisibility("visible")}
                                            value="visible"
                                            icon={<VisibilityIcon className={classes.visibilityIcon}/>}
                                            checkedIcon={<VisibilityOffIcon/>}
                                        />
                                    }
                                    label={"Make Event Secret"}

                                />
                            </FormGroup>
                        </ExpansionPanelDetails>
                    </ExpansionPanel>
                </div>
                <Button disabled={!buttonEnabled} variant="raised" color="secondary" onClick={this.handleAccept} className={classes.button}>
                    Create Event
                </Button>
            </Dialog>
        );
    }
}

CreateEventScreen.propTypes = {
    classes: PropTypes.object.isRequired,
};
export default withStyles(styles, { withTheme: true })(CreateEventScreen);