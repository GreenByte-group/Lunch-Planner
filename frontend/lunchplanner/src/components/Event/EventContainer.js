import React from 'react';
import PropTypes from 'prop-types';
import {withStyles} from '@material-ui/core/styles';
import SwipeableViews from 'react-swipeable-views';
import AppBar from '@material-ui/core/AppBar';
import {Tabs, Tab, Typography, CircularProgress} from '@material-ui/core';
import {getEvents} from "./EventFunctions";
import EventList from "./EventList";
import LocationList from "./LocationList";
import {getUsername} from "../authentication/LoginFunctions";


function TabContainer({ children, dir }) {
    return (
        <Typography component="div" dir={dir} style={{ padding: 0, height: '100%' }}>
            {children}
        </Typography>
    );
}

TabContainer.propTypes = {
    children: PropTypes.node.isRequired,
    dir: PropTypes.string.isRequired,
};

export let needReload = false;

export function eventListNeedReload() {
    needReload = true;

    if(functionToFire)
        functionToFire();
}

let functionToFire;

const styles = theme => ({
    root: {
        backgroundColor: theme.palette.background.paper,
        //width: 1500,,
        position: 'relative',
        height: '100%',
        overflowY: 'auto',
        display: 'flex',
        flexDirection: 'column',
    },
    fab: {
        position: 'absolute',
        bottom: theme.spacing.unit * 2,
        right: theme.spacing.unit * 2,
    },
    whiteSymbol: {
        color: theme.palette.common.white
    },
    tab: {
        fontFamily: "Work Sans",
        fontWeight: '600',
        letterSpacing: '0.65px',
        fontSize: '13px',
    },
    swipeViews: {
        height: '100%',
        overflowY: 'auto',
        maxWidth: '1024px',
        width: '100%',
        marginLeft: 'auto',
        marginRight: 'auto',
    },
    progress:{
        marginLeft: 'auto',
        marginRight: 'auto',
        marginTop: "auto",
        marginBottom: "auto",
        display: "block",
    }
});

class EventContainer extends React.Component {
    constructor(props) {
        super();
        this.state = {
            value: 1,
            search: props.search,
            events: [],
            loading: true,
            completed: 0,
        };

        functionToFire = () => {
            this.setState({
                loading: true,
            })
        }
    }

    componentDidMount() {
        this.setState({
            search: this.props.search,
            joinedAndInvitedEvents: [],
            loading: true,
        });

        this.loadEvents(this.props.search);

    }

    componentWillReceiveProps(newProps) {
        if(needReload) {
            needReload = !needReload;
            this.loadEvents();
        }
        if(newProps.search !== this.state.search){
            console.log(newProps.search)
            this.setState({
                search: newProps.search,
            });
            this.loadEvents(newProps.search);
        }
    }

    loadEvents(search) {
        this.setState({
            loading: true,
        });
        console.log(search);
        if(search === null || search === undefined)
            search = this.state.search;

        getEvents(search, (response) => {
            this.setState({
                events: response.data,
                loading: false,
            });
            this.sort();
        });

    }

    sort(){
        let events = this.state.events;
        let joinedAndIvitedEvents = [];
        events.forEach(function(listValue) {
            let accepted = false;
            let invited = false;
            let username = getUsername();
            listValue.invitations.forEach((value) => {
                if (value.userName === username) {
                    invited = true;
                    joinedAndIvitedEvents.push(listValue);
                    if (value.answer === 0) {
                        accepted = true;
                    }
                }
            });
        });
        this.setState({
            joinedAndInvitedEvents: joinedAndIvitedEvents,
        });
    };

    handleChange = (event, value) => {
        this.setState({ value });
    };

    handleChangeIndex = index => {
        this.setState({ value: index });
    };

    render() {
        const { classes, theme } = this.props;
        let loading = this.state.loading;
        if(needReload) {
            needReload = !needReload;
            this.loadEvents();
        }
        return (

            <div className={classes.root}>
                {loading ?
                    (this.state.events.length === 0) ?
                        <CircularProgress className={classes.progress} color="secondary"/>
                        : <CircularProgress style={{position: 'absolute', zIndex: 10000, top: 'calc(50% - 20px)', left: 'calc(50% - 20px)'}} className={classes.progress} color="secondary"/>
                    : ''
                }

                {this.state.events.length !== 0 || !loading ?
                    <div className={classes.root}>
                        <AppBar position="relative" color="default" >
                            <Tabs
                                value={this.state.value}
                                onChange={this.handleChange}
                                indicatorColor="secondary"
                                textColor="secondary"
                                centered
                                fullWidth
                            >
                                <Tab className={classes.tab} label="PERSONAL" />
                                <Tab className={classes.tab} label="PLACES" />
                                <Tab className={classes.tab} label="BY DATE" />
                            </Tabs>
                        </AppBar>
                        <SwipeableViews
                            axis={theme.direction === 'rtl' ? 'x-reverse' : 'x'}
                            index={this.state.value}
                            onChangeIndex={this.handleChangeIndex}
                            className={classes.swipeViews}
                        >

                            <TabContainer dir={theme.direction}>
                                <EventList events={this.state.joinedAndInvitedEvents}/>
                            </TabContainer>
                            <TabContainer dir={theme.direction}>
                                <LocationList events={this.state.events} />
                            </TabContainer>
                            <TabContainer dir={theme.direction}>
                                <EventList events={this.state.events} />
                            </TabContainer>
                        </SwipeableViews>
                    </div>

                    : ''
                }
            </div>

        );
    }
}

EventContainer.propTypes = {
    classes: PropTypes.object.isRequired,
    theme: PropTypes.object.isRequired,
};

export default withStyles(styles, { withTheme: true })(EventContainer);
