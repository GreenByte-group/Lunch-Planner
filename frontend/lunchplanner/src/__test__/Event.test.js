import React from 'react';
import Event from "../components/Event/Event";
import {StaticRouter} from "react-router-dom";
import { shallow, mount, render } from 'enzyme';
import * as enzyme from "enzyme";

test('Event with dummy data', () => {
    // Render a checkbox with label in the document
    const event = shallow(
        <Event background='#FFFFFF'
               id={1}
               name='event'
               description='description'
               date={new Date('2018-12-17T03:24:00')}
               people={[{userName: 'name', answer: 0, admin: true}]}
               location='location'
        />
    );

    expect(event).toMatchSnapshot();
});

