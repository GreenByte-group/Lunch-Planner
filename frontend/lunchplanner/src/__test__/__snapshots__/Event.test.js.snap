// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event with dummy data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Event)
    background="#FFFFFF"
    date={2018-12-17T02:24:00.000Z}
    description="description"
    id={1}
    location="location"
    name="event"
    people={
        Array [
            Object {
              "admin": true,
              "answer": 0,
              "userName": "name",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "background": "#FFFFFF",
      "classes": Object {
        "card": "Event-card-1",
        "cardContent": "Event-cardContent-12",
        "date": "Event-date-5",
        "footer": "Event-footer-13",
        "icons": "Event-icons-8",
        "imageDiv": "Event-imageDiv-11",
        "link": "Event-link-2",
        "listItem": "Event-listItem-3",
        "text": "Event-text-9",
        "textSelected": "Event-textSelected-10",
        "time": "Event-time-6",
        "title": "Event-title-4",
        "users": "Event-users-7",
      },
      "date": 2018-12-17T02:24:00.000Z,
      "description": "description",
      "id": 1,
      "location": "location",
      "name": "event",
      "people": Array [
        Object {
          "admin": true,
          "answer": 0,
          "userName": "name",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "background": "#FFFFFF",
        "classes": Object {
          "card": "Event-card-1",
          "cardContent": "Event-cardContent-12",
          "date": "Event-date-5",
          "footer": "Event-footer-13",
          "icons": "Event-icons-8",
          "imageDiv": "Event-imageDiv-11",
          "link": "Event-link-2",
          "listItem": "Event-listItem-3",
          "text": "Event-text-9",
          "textSelected": "Event-textSelected-10",
          "time": "Event-time-6",
          "title": "Event-title-4",
          "users": "Event-users-7",
        },
        "date": 2018-12-17T02:24:00.000Z,
        "description": "description",
        "id": 1,
        "location": "location",
        "name": "event",
        "people": Array [
          Object {
            "admin": true,
            "answer": 0,
            "userName": "name",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
